### PR TITLE
feat: vault details page

### PR DESCRIPTION
**Pull request by tyleroooo**

### CODE REVIEW

File: src/components/Table.tsx [hunk 1]
```diff
@@ -209,14 +209,16 @@ export const Table = <TableRowData extends BaseTableRowData | CustomRowConfig>({
     getKey: internalGetRowKey,
     load: async ({ sortDescriptor }) => ({
       items: sortDescriptor?.column
-        ? data.sort((a, b) => sortFn(a, b, sortDescriptor?.column, sortDescriptor?.direction))
+        ? [...data].sort((a, b) => sortFn(a, b, sortDescriptor?.column, sortDescriptor?.direction))
```
Comment (tyleroooo) [inline src/components/Table.tsx:212]: it's rude to modify your parameters

**Comment by tyleroooo** [comment]
actually there's not really any reason to merge this, all useless now

**Comment by jaredvu** [comment]
Design nits

Screen Shot 2024-07-10 at 12 35 03 PM

Spacing seems off. `<div>` wrapping the `<Output>` has larger height. Desired gap is `6px`, right now it is `4px` + `?px` (from the `<div>`.

Screen Shot 2024-07-10 at 12 36 14 PM
These `<ToggleGroups>` should be in the same row. Might be helpful to use `<Toolbar>` component.

Screen Shot 2024-07-10 at 12 40 04 PM
`var(--color-text-2)`


Screen Shot 2024-07-10 at 12 41 15 PM
Add horizontal padding to `<ToggleGroups>` in Tablet/Mobile view

**Comment by jaredvu** [comment]
Design nits

Screen Shot 2024-07-10 at 12 35 03 PM

Spacing seems off. `<div>` wrapping the `<Output>` has larger height. Desired gap is `6px`, right now it is `4px` + `?px` (from the `<div>`.

Screen Shot 2024-07-10 at 12 36 14 PM
These `<ToggleGroups>` should be in the same row. Might be helpful to use `<Toolbar>` component.

Screen Shot 2024-07-10 at 12 40 04 PM
`var(--color-text-2)`


Screen Shot 2024-07-10 at 12 41 15 PM
Add horizontal padding to `<ToggleGroups>` in Tablet/Mobile view

File: src/state/vaults.ts [hunk 2]
```diff
@@ -1,53 +1,107 @@
...
 const initialState: VaultsState = {
-  vaults: { 'PEPE-USD': { totalValue: 30_425 } },
   vaultDetails: {
-    'PEPE-USD': {
-      allTimePnl: { absolute: 4_125, percent: 0.1252 },
-      currentLeverageMultiple: 1.2,
-      currentPosition: { asset: 17341235412, usdc: 423.67 },
-      thirtyDayReturnPercent: 0.1474,
-    },
+    totalValue: 30_425,
```
Comment (yogurtandjam) [inline src/state/vaults.ts:51]: [q] is this a default value? or a placeholder for development before BE is complete

**Review by yogurtandjam — APPROVED** [review]
LGTM. we should do a full QA once we hook it up to the backend but don't see anything glaring or crazy

File: src/state/vaults.ts (hunk 2 above)
Comment (tyleroooo) [reply src/state/vaults.ts:51]: just placeholder

**Pull request merged**